### ISSUE TITLE
scanner nor recognized

### ISSUE THREAD

**Issue by melomane63** [flathub/org.gnome.OCRFeeder#2]
my hp scanner is not recognized. 
i have tried the .deb version and this recognized my scanner.

**Comment by scx** [comment]
This is a known issue and it applies to all flatpak applications. We are waiting for a proper API for accessing scanner devices since 2017.
https://bugzilla.gnome.org/show_bug.cgi?id=788831
https://github.com/flatpak/xdg-desktop-portal/issues/218

Unfortunately, there is no progress on this at all. 

There are several workarounds:
 - We can use the `net` backend, but this approach requires the user to perform some operations on the host (run the `saned` service).
 - Bundle as many backends as we can. However, it is insane to maintenance.

However, since there is not progress in providing the scanner portal, I decided to both bundle more backends and enable the `net` backend as a fallback solution.
I wanted to test this first using **XSane**, but Flathub members are still blocking this idea for no apparent reason.
https://github.com/flathub/flathub/pull/1111

That's why it took so long - I have to borrow various scanners from my friends, because I couldn't find volunteers on the Internet.
The problem is that many packages often have hardcoded paths (e.g. `/usr/` or `/etc/`, which are read-only or unavailable under flatpak) or other issues, so they require special fixes for flatpak. This was the case with `hpaio`.

I've also provided a bunch of patches for **OCRFeeder** to improve scanning support.
https://gitlab.gnome.org/GNOME/ocrfeeder/merge_requests/3

Anyway, I believe that this issue is fixed with the last update. It should be published within 24 hours.
https://github.com/flathub/org.gnome.OCRFeeder/pull/8

Here is the list of supported SANE backends:
https://github.com/flathub/org.gnome.OCRFeeder#what-sane-backends-are-supported

If you still have some problems, you can use the `net` backend.
https://github.com/flathub/org.gnome.OCRFeeder#why-sandboxed-ocrfeeder-doesnt-recognize-my-scanner-while-it-works-flawless-on-the-host

You can also build the flatpak package by yourself or install a single-file bundle (it is less restrictive when it comes to limiting permissions):
https://github.com/scx/ocrfeeder-flatpak
https://github.com/scx/ocrfeeder-flatpak/releases

In the future, we would like to use a new portal for scanning. It would probably use the **Document Scanner** (**Simple Scan**) [UI](https://blogs.gnome.org/aday/2017/05/16/new-simple-scan-designs) to return a bunch of bitmaps that we can insert as new pages in **OCRFeeder**. However, for now, it's just an idea. Nothing more, nothing less.
https://gitlab.gnome.org/GNOME/ocrfeeder/merge_requests/3#note_684563

**Comment by scx** [comment]
See also:
https://gitlab.gnome.org/GNOME/ocrfeeder/issues/9

**Comment by scx** [comment]
@melomane63 

If you're still experiencing a problem, feel free to leave a comment.

Just remember to tell me what **SANE** backend your scanner is using. You can check it this way:
```
scanimage -L
```

**Comment by Alexander-Wilms** [comment]
I get the error message "No scanner devices were found". Non-flatpak'd Simple Scan and Skanlite work fine.

```
scanimage -L
device `brother4:net1;dev0' is a Brother MFC-J6510DW MFC-J6510DW
```
```
flatpak info org.gnome.OCRFeeder

OCRFeeder - The complete OCR suite

          ID: org.gnome.OCRFeeder
         Ref: app/org.gnome.OCRFeeder/x86_64/stable
        Arch: x86_64
      Branch: stable
     Version: 0.8.3
     License: GPL-3.0+
      Origin: flathub
  Collection: org.flathub.Stable
Installation: system
   Installed: 2.4?GB
     Runtime: org.gnome.Platform/x86_64/42
         Sdk: org.gnome.Sdk/x86_64/42

      Commit: da7545fb4193e708b37cda4aaf5b77a8c58985e81e2f3dc2847c79f7608cb247
      Parent: 31aa7380b7a71e3e6964e266d39c0d79c61eb1a088bc48eb98299d4a32396903
     Subject: Fix gvfs dbus (d7267c62)
        Date: 2022-05-31 03:25:48 +0000
```
